test/e2e/datafederation_pe_test.go

```diff
@@ -127,5 +130,24 @@ var _ = Describe("UserLogin", Label("datafederation"), func() {
 			}, df)).To(Succeed())
 			Expect(testData.K8SClient.Delete(testData.Context, df)).Should(Succeed())
 		})
+
+		By("Deleting DataFederation Private Endpoint", func() {
+			// This is required or will result on error:
+			// CANNOT_CLOSE_GROUP_ACTIVE_ATLAS_DATA_FEDERATION_PRIVATE_ENDPOINTS
+			// for some reason, doing it just once doesn't work
+			for i := 0; i < 5; i++ { // 3 retries or less seems to fail
+				_, resp, err := atlasClient.Client.DataFederationApi.DeleteDataFederationPrivateEndpoint(testData.Context,
+					testData.Project.ID(), pe.ID).Execute()
+				if admin.IsErrorCode(err, "RESOURCE_NOT_FOUND") {
+					msg := fmt.Sprintf("It says it Removed DataFederation Private Endpoint %q for project %s after retry %d\n",
+						pe.ID, testData.Project.ID(), i+1)
+					GinkgoWriter.Write([]byte(msg))
+					continue
+				}
```

Comment:
non-blocking: @josvazg you can replace it with `Eventually()`

Reply:
I would rather not change that. `Eventually` is used when you have a eventually consistent state to check for, which is not the case here. I would instead add a comment linking to the Atlas issue so we can revisit and fix this code later on, removing the for completely and, most likely not needing even to use `Eventually` either.